src/Pim/Component/Catalog/Model/AbstractCompleteness.php

```diff
@@ -14,76 +18,83 @@
     /** @var int|string */
     protected $id;
 
+    /** @var ProductInterface */
+    protected $product;
+
     /** @var LocaleInterface */
     protected $locale;
 
     /** @var ChannelInterface */
     protected $channel;
 
     /** @var int */
-    protected $ratio = 100;
+    protected $ratio;
 
     /** @var int */
-    protected $missingCount = 0;
+    protected $missingCount;
 
     /** @var int */
-    protected $requiredCount = 0;
-
-    /** @var ProductInterface */
-    protected $product;
+    protected $requiredCount;
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getLocale()
-    {
-        return $this->locale;
-    }
+    /** @var Collection */
+    protected $missingAttributes;
 
     /**
-     * {@inheritdoc}
+     * @param ProductInterface $product
+     * @param ChannelInterface $channel
+     * @param LocaleInterface  $locale
+     * @param Collection       $missingAttributes
+     * @param int              $missingCount
+     * @param int              $requiredCount
      */
-    public function setLocale(LocaleInterface $locale)
-    {
+    public function __construct(
```

Comment:
no it's not a value object
completeness has an identity (product + channel + locale), so it can't be a value object
here it's just an object without setters

I don't know if good or bad yet, but (it works)  we can try to see what happens with that
normally, for an entity, we should get the entity and modify only what we have to modify via the setters (required count and missing count)